### PR TITLE
[Platform API][Fan][Fan Drawer] add fan tests for fan drawers and compare applicable values against platform.json

### PR DESCRIPTION
**Pull request by vdahiya12**
Summary:
Since fans will be part of fan_drawers this PR  adds fan tests  which belong to fan drawers
For fan drawers's fan tests where known facts are available in the platform.json file, compare actual values against those truths
### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?

To improve the quality of the Fan_Drawer platform API test

#### How did you do it?
Added all tests which belong to fans in fan drawers
For fan tests where known facts are available in the platform.json file, compare actual values against those truths

#### How did you verify/test it?

Run Fan Drawer fan tests on a device with a populated platform.json present

#### Any platform specific information?

Platform must have a populated platform.json present

Signed-off-by: vaibhav-dahiya <vdahiya@microsoft.com>

### CODE REVIEW

File: tests/platform_tests/api/test_fan_drawer_fans.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, fan_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expected_fan_drawers = self.chassis_facts.get("fan_drawer")
+            if expected_fan_drawers:
+                expected_fans = expected_fan_drawers[fan_idx/2].get("fans")
```
Comment (jleveque) [inline tests/platform_tests/api/test_fan_drawer_fans.py:74]: We can't assume every fan drawer has two fans. The fan_drawer index should be passed into this function instead.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/platform_tests/api/test_fan_drawer_fans.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+    # This fixture would probably be better scoped at the class level, but
+    # it relies on the platform_api_conn fixture, which is scoped at the function
+    # level, so we must do the same here to prevent a scope mismatch.
+
+    @pytest.fixture(scope="function", autouse=True)
+    def setup(self, platform_api_conn):
+        if self.num_fans is None:
+            try:
+                self.num_fans = int(fan_drawer.get_num_fans(platform_api_conn))
```
Comment (jleveque) [inline tests/platform_tests/api/test_fan_drawer_fans.py:60]: Shouldn't need to cast to int. The fan_drawer tests should ensure it returns an int.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/platform_tests/api/test_fan_drawer_fans.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                       "Unable to get expected value for '{}' from platform.json file for fan {}".format(key, fan_idx)):
+            self.expect(value == expected_value,
+                          "'{}' value is incorrect. Got '{}', expected '{}' for fan {}".format(key, value, expected_value, fan_idx))
+
+
+    #
+    # Functions to test methods inherited from DeviceBase class
+    #
+    def test_get_name(self, duthost, localhost, platform_api_conn):
+        for i in range(self.num_fans):
```
Comment (jleveque) [inline tests/platform_tests/api/test_fan_drawer_fans.py:88]: These functions should have an outer loop which loops through all fan drawers.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/platform_tests/api/test_fan_drawer_fans.py (hunk 3 above)
Comment (vdahiya12) [reply tests/platform_tests/api/test_fan_drawer_fans.py:88]: was thinking about this, 
Thanks for the clarification

Comment (vdahiya12) [reply tests/platform_tests/api/test_fan_drawer_fans.py]: fixed

File: tests/platform_tests/api/test_fan_drawer_fans.py (hunk 1 above)
Comment (vdahiya12) [reply tests/platform_tests/api/test_fan_drawer_fans.py]: fixed

File: tests/platform_tests/api/test_fan_drawer_fans.py (hunk 2 above)
Comment (vdahiya12) [reply tests/platform_tests/api/test_fan_drawer_fans.py]: fixed

**Review by jleveque — APPROVED** [review]

**Pull request merged**